tests/test_data_integrity.py

```diff
@@ -158,8 +158,12 @@ def validator(data):
             if "datadog-entity-id" not in request_headers:
                 raise ValueError(f"Datadog-Entity-ID header is missing in request {data['log_filename']}")
             val = request_headers["datadog-entity-id"]
-            assert val.startswith("in-"), f"Datadog-Entity-ID header value {val} doesn't start with 'in-'"
-            assert val[3:].isdigit(), f"Datadog-Entity-ID header value {val} doesn't end with digits"
+            if val.startswith("in-"):
+                assert val[3:].isdigit(), f"Datadog-Entity-ID header value {val} doesn't end with digits"
+            elif val.startswith("cid-"):
+                assert val[4:].isxdigit(), f"Datadog-Entity-ID header value {val} doesn't end with hex digits"
```

Comment:
In this CI, only the first use case `val.startswith("in-")` is seen (I don't know why.

For the second use case `isxdigit` is not a real string method. Did you meant `isdigit` @ygree ?

Reply:
And I don't understand why it's green on circleCI here : https://app.circleci.com/pipelines/github/DataDog/system-tests/11929/workflows/1240bf05-d33f-498f-ba52-0d56abedb53e/jobs/11923

Is it possible that sometimes it's `in-`, and sometimes `cid-` ?

Reply:
Fixing it in #2125